lib/smooth/application/display/DisplaySpi.cpp

```diff
@@ -59,87 +58,146 @@ namespace smooth::application::display
                         use_pre_transaction_callback,
                         use_post_transaction_callback),
 
-              reset_pin(reset_pin, true, false, false),
-              backlight_pin(back_light_pin, true, false, false),
-              dc_pin(data_command_pin, true, false, false),
-              cs_pin(chip_select_pin, true, false, false)
+              reset_pin(reset_pin, true, false, false),     // GPIO_MODE_OUTPUT, no pullup, no pulldown
+              dc_pin(data_command_pin, true, false, false), // GPIO_MODE_OUTPUT, no pullup, no pulldown
+              cs_pin(chip_select_pin, true, false, false)   // GPIO_MODE_OUTPUT, no pullup, no pulldown
     {
     }
 
     // Initialize the display
-    bool ILI9341::init(spi_host_device_t host)
+    bool DisplaySpi::init(spi_host_device_t host)
     {
+        reset_pin.set(PIN_HIGH);
+        cs_pin.set(PIN_HIGH);
+
         // spi_transaction will not control chip select
         return initialize(host, GPIO_NUM_NC);
     }
 
-    // I have found different displays set the value of madctl differently
-    // for the same rotation.  For example M5Stack sets madctl to 0x68 for
-    // portrait while the adafruit 2.88 dipslay sets madctl to 0x48.  To make
-    // driver more versatile I have used madctl_value instead of the typical
-    // 0-3 screen_rotation_value that is used in switch statement.
+    // Most display ICs use MADCTL to set screen orientation. Typically there
+    // are four different screen orientations - Portrait, Landsacpe,
+    // Portrait-Flipped and Landscape-Flipped.  The problem is there is not
+    // a universal standard for setting the MADCTL value for a particular
+    // screen oreintation.  For example M5Stack uses a MADCT value of 0x68
+    // for portrait while the Adafruit 2.88 dipslay uses a MADCTL value of 0x48.
+    // To make driver more versatile the function set_madctl is used to allow user
+    // to set the screen orientation instead of the typical 0-3
+    // screen_rotation_constant that is used in switch statement.
     // If you are using LittlevGL then you need to set the appropriate screen
     // width and screen height for the rotation you have chosen in the lvconf.h file.
-    bool ILI9341::set_rotation(uint8_t madctl_value)
+    bool DisplaySpi::set_madctl(uint8_t value, uint8_t madctl_reg)
     {
-        bool res = lcd_wr_cmd(0x36);
-        res &= lcd_wr_data(&madctl_value, 1);
+        bool res = send_cmd(madctl_reg);
+        res &= send_data(&value, 1);
 
         return res;
     }
 
-    // Reset the display
-    void ILI9341::reset_display()
+    // Hardware Reset
+    void DisplaySpi::hw_reset(bool active_low, int active_time, int delay_time)
```

Comment:
Pass in `std::chrono::milliseconds` instead of an int for the time-parameters, it makes the API more clear.

Reply:
Done